### PR TITLE
add impl from anyvalue for literal

### PR DESCRIPTION
**Pull request by savente93**
As per the discussion on discord, I've added an impl to convert `AnyValue` into a `Literal`. I've tried to make sensible decisions if variants didn't have a clear analogue but am open to change them if people think they should be done differently. Also clippy did have some complaints but not about any of the code that I touched so I decided not to mess with it. Hopefully this will be an easy merge.

### CODE REVIEW

File: polars/polars-lazy/src/logical_plan/lit.rs [hunk 1]
```diff
@@ -125,6 +125,50 @@ impl<'a> Literal for &'a str {
...
+                    Self::Duration(ChronoDuration::microseconds(chrono_duration), TimeUnit::Microseconds)
+                }
+                TimeUnit::Milliseconds => {
+                    Self::Duration(ChronoDuration::milliseconds(chrono_duration), TimeUnit::Milliseconds)
+                }
+            },
+            AnyValue::Time(nano_secs_sinds_midnight) => Self::Int64(nano_secs_sinds_midnight),
+            AnyValue::List(l) => Self::Series(SpecialEq::new(l)),
+            AnyValue::Utf8Owned(o) => Self::Utf8(o),
+            AnyValue::Categorical(c, _rev_mapping) => Self::UInt32(c),
```
Comment (ritchie46) [inline polars/polars-lazy/src/logical_plan/lit.rs:166]: I think this should return a `Utf8` literal.

File: polars/polars-lazy/src/logical_plan/lit.rs [hunk 2]
```diff
@@ -125,6 +125,50 @@ impl<'a> Literal for &'a str {
...
+            _ => Self::Null
```
Comment (ritchie46) [inline polars/polars-lazy/src/logical_plan/lit.rs:167]: maybe panic on unknown types? That will be much easier to debug later.

Comment (savente93) [reply polars/polars-lazy/src/logical_plan/lit.rs:167]: Wouldn't it be better then to implement the `TryFrom` and return an Error? That way we could maybe provide some more context.

File: polars/polars-lazy/src/logical_plan/lit.rs (hunk 1 above)
Comment (savente93) [reply polars/polars-lazy/src/logical_plan/lit.rs:166]: good call

**Comment by savente93** [comment]
Sorry for all the failed CIs, took me a while to figure out that I forgot conditional compilations, but it should be good now (I hope)

**Comment by ritchie46** [comment]
> Sorry for all the failed CIs, took me a while to figure out that I forgot conditional compilations, but it should be good now (I hope)

No worries. Thanks for the PR.

**Pull request merged**